server/src/main/java/com/epam/aidial/core/server/service/ApplicationService.java

```diff
@@ -270,9 +270,11 @@ public void deleteApplication(ResourceDescriptor resource, EtagHeader etag) {
 
         Application application = reference.getValue();
 
+        boolean deleted = true;
         if (isPublicOrReview(resource) && application.getFunction() != null) {
-            deleteFolder(application.getFunction().getSourceFolder());
+            deleted = deleteFolder(application.getFunction().getSourceFolder());
         }
+        return deleted;
```

Comment:
not correct.

Reply:
I guess you want to throw exception instead of return a Boolean result or what?

Reply:
Line 259 throws an exception if an application does not exist. This code just deletes the app folder.

Reply:
fixed